llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -5939,6 +5956,9 @@ const Value *llvm::getUnderlyingObject(const Value *V, unsigned MaxLookup) {
           V = RP;
           continue;
         }
+      } else if (auto *AIEObject = getUnderlyingObjectAIEIntrinsic(V)) {
```

Comment:
Could you summarize how that differs from our previous approach of having a separate AA pass for AIE intrinsics?

Reply:
Hi @gbossu, basically, our pass can handle just linear code, in this way, we can decompose only in upwards direction. There is no simple way to, for example, switch between different analysis implementation when we don't understand a specific node (PHI, for example). Once we are running the BasicAA, and we reach a PHI node, for example, we cannot easily decide which AA in the chain can handle all the incoming values from that node. 

One approach could be the propagation of the PHI handling inside our AA, however we will definitely handle PHI chains twice (AIEAA an BasicAA)  for the cases containing our specific intrinsics.

Reply:
I feel that it is better to have one heavy AA that can handle everything than two heavy AAs. If the first fail to prove NoAlias, we need to run everything again (next AA).

Reply:
Let's delay the removal of our custom AA wrapper pass. We might anyway need it for address space disambiguation.